GenericTensor/Declaration.cs

```diff
@@ -67,6 +67,20 @@ public sealed partial class GenTensor<T, TWrapper>
         public static GenTensor<T, TWrapper> Concat(GenTensor<T, TWrapper> a, GenTensor<T, TWrapper> b)
             => Composition<T, TWrapper>.Concat(a, b);
 
+
+        /// <summary>
+        /// Folds the tensor over given axis
+        /// axis : int - the axis' number
+        /// collapse : (acc : T) x (step : T) -> unit
+        /// acc : T[N1 x ... x N[axis-1] x N[axis+1] x ... x Nn]
+        /// t : T[N1 x N2 x N3 x ... Nn]
+        /// Modifies acc
+        /// </summary>
+        public static void Fold<TCollapse, U, UWrapper>(TCollapse collapse, GenTensor<U, UWrapper> acc, GenTensor<T, TWrapper> t, int axis)
```

Comment:
Might be better to follow names defined by https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.aggregate?view=net-6.0#system-linq-enumerable-aggregate-2(system-collections-generic-ienumerable((-0))-1-system-func((-1-0-1))) since you are exposing to C# users

Also you should indicate that `TCollapse` is a function by adding the word `Func` or equivalent

Reply:
Not solved yet

Reply:
specifically `TAccumulate seed`